version: '3.9'

services:
  vtx:
    image: ghcr.io/0-5788719150923125/bot:latest
    command: nodemon --ext '*.py, *.yml' --ignore "models/*" --ignore "*.json" --exec python3 main.py
    restart: 'always'
    build:
      context: .
      dockerfile: Dockerfile
      shm_size: '1gb'
    stdin_open: true
    tty: true
    volumes:
      - ./gen:/gen
      - ./lab:/lab
      - ./vtx:/vtx
    env_file:
      - .env
    environment:
      FOCUS: ${FOCUS:-heart}
      TRANSFORMERS_CACHE: /tmp

  ctx:
    image: ghcr.io/0-5788719150923125/src:latest
    restart: 'always'
    ports:
      - 9666:9666
    environment:
      WEBUI: disabled
      ANONYMOUS: 'true'

  pit:
    image: ghcr.io/0-5788719150923125/bot:latest
    command: tensorboard --logdir /gen/logs --bind_all --samples_per_plugin scalars=999999999
    volumes:
      - ./gen:/gen
    ports:
      - 6006:6006

  pet:
    image: learningathome/petals:main
    command: python -m petals.cli.run_server bigscience/bloom-560m --cache_dir /vtx/models --num_blocks 24 --torch_dtype float32
    restart: 'always'
    network_mode: 'host'
    ipc: host
    deploy:
      resources:
        limits:
          cpus: '0.75'
    volumes:
      - ./vtx/models:/vtx/models
    environment:
      HIVEMIND_LOGLEVEL: ERROR